test/hotspot/jtreg/serviceability/jvmti/RedefineClasses/RedefineVerifyError.java

```diff
@@ -0,0 +1,130 @@
+
+        classWriter.visit(52, ACC_SUPER | ACC_PUBLIC, "java/lang/VerifyError", null, "java/lang/LinkageError", null);
+        {
+            fieldVisitor = classWriter.visitField(ACC_PRIVATE | ACC_FINAL | ACC_STATIC, "serialVersionUID", "J", null, new Long(7001962396098498785L));
+            fieldVisitor.visitEnd();
+        }
+        {
+            methodVisitor = classWriter.visitMethod(ACC_PUBLIC, "<init>", "()V", null, null);
+            methodVisitor.visitCode();
+            Label label0 = new Label();
+            methodVisitor.visitLabel(label0);
+            methodVisitor.visitLineNumber(43, label0);
+            methodVisitor.visitVarInsn(ALOAD, 0);
+            methodVisitor.visitMethodInsn(INVOKESPECIAL, "java/lang/LinkageError", "<init>", "()V", false);
+            Label label1 = new Label();
+            methodVisitor.visitLabel(label1);
+            methodVisitor.visitLineNumber(44, label1);
+            methodVisitor.visitInsn(RETURN);
+            Label label2 = new Label();
+            methodVisitor.visitLabel(label2);
+            methodVisitor.visitLocalVariable("this", "Ljava/lang/VerifyError;", null, label0, label2, 0);
+            methodVisitor.visitMaxs(1, 1);
+            methodVisitor.visitEnd();
+        }
+
+        {   // broken method
+            methodVisitor = classWriter.visitMethod(ACC_PUBLIC, "<init>", "(Ljava/lang/String;)V", null, null);
+            methodVisitor.visitCode();
+            Label label0 = new Label();
+            Label label1 = new Label();
+            Label label2 = new Label();
+            methodVisitor.visitTryCatchBlock(label0, label1, label2, "java/lang/Exception");
+            methodVisitor.visitLabel(label0);
+            methodVisitor.visitVarInsn(ALOAD, 0);
+            methodVisitor.visitVarInsn(ALOAD, 1);
+            methodVisitor.visitMethodInsn(INVOKESPECIAL, "java/lang/LinkageError", "<init>", "(Ljava/lang/String;)V", false);
+            methodVisitor.visitLabel(label1);
+            Label label3 = new Label();
+            methodVisitor.visitJumpInsn(GOTO, label3);
+            methodVisitor.visitLabel(label2);
+            methodVisitor.visitFrame(Opcodes.F_FULL, 2, new Object[] {"MyError", "java/lang/String"}, 1, new Object[] {"java/lang/Exception"});
+            methodVisitor.visitVarInsn(ASTORE, 2);
+            methodVisitor.visitLabel(label3);
+            // methodVisitor.visitFrame(Opcodes.F_SAME, 0, null, 0, null);
+            methodVisitor.visitInsn(RETURN);
+            methodVisitor.visitMaxs(2, 3);
+            methodVisitor.visitEnd();
+            classWriter.visitEnd();
+        }
+
+        return classWriter.toByteArray();
+    }
+
+    public static void main(String[] args) throws Exception {
+
+        Class<?> verifyErrorMirror = java.lang.VerifyError.class;
+
+        try {
+            RedefineClassHelper.redefineClass(verifyErrorMirror, dump());
+            VerifyError err = new VerifyError("verify me now");
```

Comment:
Which of these two lines are expected to throw a VerifyError?

Is the VerifyError because you can't redefine VerifyError, or is there something invalid about the how the class is redefined?

Reply:
We call the verifier during redefinition with the 'true' parameter, so the redefinition will throw VerifyError when it's loading the new class file version.  There's a broken stackmap in the new version of this VerifyError class (ie. I asmifyied it and commented out something in it).  I updated the test with some comments.

Reply:
Ok. You got rid of the `new VerifyError()`, which is what I figured my next request was going to be. :)

The test looks fine. I don't really have the background to review the native code.